package.json

```diff
@@ -36,7 +36,7 @@
     "pep8": "pycodestyle --max-line-length 120 -r script",
     "pylint": "node ./build/commands/scripts/commands.js pylint",
     "web-ui-gen-grd": "node components/webpack/gen-webpack-grd",
-    "web-ui-gen-tsconfig": "node components/webpack/gen-tsconfig",
+    "web-ui-gen-tsconfig": "node build/commands/scripts/genTsConfig",
```

Comment:
Can we make `web-ui-gen-tsconfig` and `build-storybook` take the `target_arch`, `target_os` and`target_environment` as parameters, which then get forwarded via the environment variables or some other way? We have `npm run build Component` so `npm run build-storybook Component` makes intuitive sense. Similarly for `npm run build -- Static --target_arch=arm64` and `target_os`.

Actually, in my opinion, it seems like the storybook should be a GN target. Then we would automatically get all the target_arch etc. logic for free. But it may be beyond the scope of this PR.

Reply:
There is a storybook GN target which ensures the dependencies are built.
I do not think running storybook via GN is necessarily the best or obvious approach. Storybook is not an artifact of brave-core but a tool that uses the artifacts from brave-core as 1 of its dependencies. It creates a long running web server with a watch on all dependencies.

As far as I know, the `storybook` and `build-storybook` commands do not expose their cli parameters to their configurations. We would have to write wrapper commands to accept cli parameters. I used environment variables for now since this is a CI-only issue and CI needs to be unblocked without waiting for someone to have the time to change everything about how we use storybook.

`web-ui-gen-tsconfig` is already only internally run from GN. It's also not called from this storybook config, only the browser build. I just refactored it so it can be called as a function. It doesn't need to accept parameters because it isn't run directly by the user or CI.

Reply:
> CI needs to be unblocked without waiting for someone to have the time to change everything about how we use storybook. It can be done as a follow-up

I understand and agree; The priority is to unblock CI. We can do additional work in a follow-up.

> Storybook is not an artifact of brave-core but a tool that uses the artifacts from brave-core as 1 of its dependencies.

In my opinion, the fact that Storybook has GN dependencies and itself needs `target_os` etc. indicates to me that GN is a natural place for it to be called from.

> this is a CI-only issue

I also experienced the issue during local development, so it's not a CI-only issue in my opinion.